### ISSUE TITLE
Custom MessageConverter is not being used when Spring Websocket is also used

### ISSUE THREAD

**Issue by jarebudev** [spring-cloud/spring-cloud-stream#1920]
I first asked this on Stackoverflow and was requested to raise an issue here.  

https://stackoverflow.com/questions/60470262

I have a Scs microservice that consumes messages from a series of Kafka topics which all share the same abstract class.

I've registered a custom `MessageConverter` to deserialise the JSON messages that use this abstract class.

I wanted to push these messages I receive from Kafka through a websocket to our front end UI. Once I added the config for the websocket broker the MessageConverter for Scs was no longer being used. It looks like the MessageConverter's registered for use with the websocket is taking precedence over the one that I've registered for use with Scs.

I've created a cut down example here https://github.com/jarebudev/scs-websocket-messageconverter-demo to demonstrate the issue - see the provided unit test.

The version I'm using is `Hoxton.SR2`

**Comment by jarebudev** [comment]
Ok so I just had a play about with this and managed to work around this problem for now.

My `@Configuration' beans are in the same package.

`com.example.demo.config.AppWebsocketConfig`
`com.example.demo.config.ScsConfig`

I renamed `AppWebsocketConfig`  to `WebsocketConfig` and this is registering the `ScsConfig` first and enabling my `CustomMessageConverter` to be used.

see branch here https://github.com/jarebudev/scs-websocket-messageconverter-demo/tree/reverse_config_order

**Comment by olegz** [comment]
Actually your sample app doesn't start
```
Action:

Consider defining a bean of type 'com.example.demo.service.DemoService' in your configuration.

```

Also, I don't see any custom converters being bootstrapped. I'll dig some more. . .

**Comment by olegz** [comment]
@jarebudev you should also simplify your app by embracing functional programming style.
Basically change your `DemoApplication` to  look like this:
```
@SpringBootApplication
public class DemoApplication {

	public static void main(String[] args) {
		SpringApplication.run(DemoApplication.class, args);
	}

	@Autowired
	private DemoService demoService;


        public Consumer<AbstractObject> handler() {
              return message -> {
                   System.out.println("received message, commonValue = " + 
                        message.getCommonValue() + ", message = " + message.toString());
		   // service that will send message into websocket...
		   demoService.handle(message);
              };
        }
}
```

**Comment by jarebudev** [comment]
@olegz thanks, yes moving away from annotation-based to functional is on our backlog

Thanks for investigating and the prompt fix!

**Comment by davidwadden** [comment]
@olegz I just bumped into this one attempting to upgrade from Hoxton.SR2 to Hoxton.SR3.  My app broke that was relying on a [`MarshallingMessageConverter`](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/messaging/converter/MarshallingMessageConverter.html) bean to marshal JAXB object hierarchies to XML in the message converter layer.    I tracked the change to [`ContentTypeConfiguration.java:66`](https://github.com/spring-cloud/spring-cloud-stream/blob/v3.0.4.RELEASE/spring-cloud-stream/src/main/java/org/springframework/cloud/stream/config/ContentTypeConfiguration.java#L66) filtering this converter out for `application/xml`.

As a short-term workaround, I can extend the MessageConverter into an alternate namespace to circumvent this check, but wanted to see if you had a recommendation for this configuration.
